stdlib/public/Platform/Platform.swift

```diff
@@ -286,11 +313,24 @@ public var S_IREAD: mode_t  { return S_IRUSR }
 public var S_IWRITE: mode_t { return S_IWUSR }
 public var S_IEXEC: mode_t  { return S_IXUSR }
 #endif
+#else
+public var S_IFMT: Int32 { return Int32(0xf000) }
+
+public var S_IFREG: Int32 { return Int32(0x8000) }
+public var S_IFDIR: Int32 { return Int32(0x4000) }
+public var S_IFCHR: Int32 { return Int32(0x2000) }
+public var S_IFIFO: Int32 { return Int32(0x1000) }
+
+public var S_IREAD: Int32  { return Int32(0x0100) }
+public var S_IWRITE: Int32 { return Int32(0x0080) }
+public var S_IEXEC: Int32  { return Int32(0x0040) }
```

Comment:
Windows does not have `mode_t`?  This is painful.

The definitions look otherwise identical.  We actually also have `S_IFMT`, `S_IFREG`, `S_IFDIR`, `S_IFCHR`, `S_IFIFO` on Apple platforms.

Could we gyb this somehow?  Or maybe emulate `mode_t` on Windows?